app/api/update_houses.php

```diff
@@ -146,6 +148,21 @@ function fetch_hemnet_houses(){
         // Array keys two-deep
         $this_house['lat'] = $listing['coordinates']['lat'];
         $this_house['lng'] = $listing['coordinates']['long'];
+
+        // Check if the coordinates already exist in the $houses array
+        $lat_lng_key = $this_house['lat'] . ',' . $this_house['lng'];
+        if (isset($all_lat_lng[$lat_lng_key])) {
+            // Coordinates already exist, adjust them slightly to move the house 5 meters away
+
+            $newCoordinates = moveHouseCoordinates($this_house['lat'], $this_house['lng'], 5); // 5 meters
+
+            $this_house['lat'] = $newCoordinates['lat'];
+            $this_house['lng'] = $newCoordinates['lng'];
+        }
+        $lat_lng_key = $this_house['lat'] . ',' . $this_house['lng'];  // Update key with possibly new coordinates
```

Comment:
Not sure how likely it is to happen, but what about putting this into a `while` loop in case the new coordinates are also taken?

```suggestion
        while (isset($all_lat_lng[$lat_lng_key])) {
            // Coordinates already exist, adjust them slightly to move the house 5 meters away

            $newCoordinates = moveHouseCoordinates($this_house['lat'], $this_house['lng'], 5); // 5 meters

            $this_house['lat'] = $newCoordinates['lat'];
            $this_house['lng'] = $newCoordinates['lng'];
            $lat_lng_key = $this_house['lat'] . ',' . $this_house['lng'];  // Update key with possibly new coordinates
        }
```